src/app/ad-authoring/generate-amp-html.ts

```diff
@@ -0,0 +1,202 @@
+export function generateAdAmpHtml(
+  callToActionStr: string,
+  landingPageUrl: string,
+  landingPageType: string,
+  base64AssetStr: string,
+  assetFile: File
+) {
+  let adAMPHTML =
+    '"<!doctype html><html amp4ads><head><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,minimum-scale=1\\"><meta name=\\"amp-cta-type\\" content=\\"' +
+    callToActionStr +
+    '\\"><meta name=\\"amp-cta-url\\" content=\\"' +
+    landingPageUrl +
+    '\\"><meta name=\\"amp-cta-landing-page-type\\" content=\\"' +
+    landingPageType +
```

Comment:
Let's factor out this part of the HTML into a const and add the image/video based on the file type (similar to how you're doing in the downloadService).

Reply:
This is taken care of in the downloadZip PR.